### PR TITLE
Merge US007-8 -> Main

### PR DESCRIPTION
**Pull request by AurorinhaBoreal**
This pull request circles primarly on Unit Testing and test configuration. Again. this will be made on [GitLab](https://gitlab.dbserver.com.br/aurora.rossi/personcrud).

As it is today, this branch has Jacoco Configuration (that will be descontinued because it is unnecessary) and mostly tests around PersonService.

This branch will have more tests soon, that will circle around AddressService, AddressController and PersonController.

Again, in GitLab I will link the US's and everything else related to this.

### CODE REVIEW

File: src/main/java/com/db/crud/person/controller/PersonController.java [hunk 1]
```diff
@@ -21,29 +27,52 @@
...
+        return personService.findAll(pageable);
+    }
+    
+
     @PostMapping("/create")
-    public void createUser(@RequestBody @Valid PersonDTO person) {
-        service.verifyCPF(person.getCpf());
-        service.create(new Person(person));
+    public ResponseEntity<Person> createUser(@RequestBody @Valid PersonDTO person) {
+        personService.verifyCPF(person.getCpf());
```
Comment (Mateusun1) [inline src/main/java/com/db/crud/person/controller/PersonController.java:54]: poderia transformar esse método aqui em private e puxar ele apenas dentro da Service

File: src/main/java/com/db/crud/person/controller/PersonController.java [hunk 2]
```diff
@@ -21,29 +27,52 @@
...
         log.info("Atualizando Pessoa: "+person);
-        service.update(person, personID);
+        var info = personService.update(person, personID);
+        return ResponseEntity.status(HttpStatus.OK).body(info);
     }
+
+    @DeleteMapping("/delete/{personID}")
+    public ResponseEntity<String> deleteUser(@PathVariable Long personID) {
+        personService.delete(personID);
+        return ResponseEntity.status(HttpStatus.NO_CONTENT).body("A pessoa com ID "+personID+" foi deletada com sucesso!");
```
Comment (Mateusun1) [inline src/main/java/com/db/crud/person/controller/PersonController.java:69]: Acho que na teoria, quando o Status é NO_CONTENT, não precisa retornar nada no Body

File: src/main/java/com/db/crud/person/entity/Person.java [hunk 3]
```diff
@@ -45,7 +45,7 @@ public class Person {
     @NotNull(message = "Informe uma data válida!")
     private LocalDate birthDate;
 
-    @OneToMany(mappedBy = "personID", cascade = CascadeType.PERSIST) // Identifica a classe pessoa como 1 para muitas em relação a endereço (1:n)
+    @OneToMany(mappedBy = "personID", cascade = CascadeType.ALL) // Identifica a classe pessoa como 1 para muitas em relação a endereço (1:n)
```
Comment (Mateusun1) [inline src/main/java/com/db/crud/person/entity/Person.java:48]: Não esqueça de remover os comentarios, não é boa prática

File: src/main/java/com/db/crud/person/mapper/AddressDTOMapper.java [hunk 4]
```diff
@@ -1,5 +1,19 @@
-package com.db.crud.person.mapper;
+// package com.db.crud.person.mapper;
```
Comment (Mateusun1) [inline src/main/java/com/db/crud/person/mapper/AddressDTOMapper.java:1]: remova essa classe por um momento, quando você iniciar uma nova branch que contenha os mappers completos, vc utiliza ele :)

File: src/main/java/com/db/crud/person/mapper/PersonDTOMapper.java
Comment (Mateusun1) [inline src/main/java/com/db/crud/person/mapper/PersonDTOMapper.java:1]: mesma coisa com esse daqui

**Review by Mateusun1 — CHANGES_REQUESTED** [review]

File: src/main/java/com/db/crud/person/controller/PersonController.java (hunk 2 above)
Comment (AurorinhaBoreal) [reply src/main/java/com/db/crud/person/controller/PersonController.java]: Feedback resolvido no commit [
refacotr: Change how to Handle Delete Methods](https://github.com/AurorinhaBoreal/PersonCRUD-Temp/pull/4/commits/a15cb5c5b896fa9eb3065c40f0480992a35025f7)

File: src/main/java/com/db/crud/person/controller/PersonController.java (hunk 1 above)
Comment (AurorinhaBoreal) [reply src/main/java/com/db/crud/person/controller/PersonController.java]: Feedback implementado no commit [Change When to VerifyCpf](https://github.com/AurorinhaBoreal/PersonCRUD-Temp/pull/4/commits/670fa2ac504f6621034a3b364796a0cc8080f057)

File: src/main/java/com/db/crud/person/mapper/PersonDTOMapper.java
Comment (AurorinhaBoreal) [reply src/main/java/com/db/crud/person/mapper/PersonDTOMapper.java:1]: Feedback também implmentado no [chore: Remove Unused Files and Dependecies](https://github.com/AurorinhaBoreal/PersonCRUD-Temp/pull/4/commits/807f695e6038efaa276de664bae9a608d2c24404)

File: src/main/java/com/db/crud/person/entity/Person.java (hunk 3 above)
Comment (AurorinhaBoreal) [reply src/main/java/com/db/crud/person/entity/Person.java]: Feedback Implementado no Commit [refactor: Remove Comentaries and Unused Dependencies](https://github.com/AurorinhaBoreal/PersonCRUD-Temp/pull/4/commits/96b89ff671190fa9e6e21bcc487c0680fe3a31e1)

**Pull request merged**